src/runtime/virtcontainers/qemu.go

```diff
@@ -369,13 +374,30 @@ func (q *qemu) createQmpSocket() ([]govmmQemu.QMPSocket, error) {
 		path: monitorSockPath,
 	}
 
-	return []govmmQemu.QMPSocket{
-		{
+	var sockets []govmmQemu.QMPSocket
+
+	if q.HypervisorConfig().Debug {
+		humanMonitorSockPath, err := q.hmpSocketPath(q.id)
+		if err != nil {
+			return nil, err
+		}
+
+		sockets = append(sockets, govmmQemu.QMPSocket{
 			Type:   "unix",
+			IsHmp:  true,
+			Name:   humanMonitorSockPath,
 			Server: true,
 			NoWait: true,
-		},
-	}, nil
+		})
+	}
+
+	sockets = append(sockets, govmmQemu.QMPSocket{
```

Comment:
Here  https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/qemu.go#L865 qemu assumed the first qmp socket device as the "QMP" interface, thus you should make sure append it first. Otherwise, qemu would use the hmp device as the "QMP" interface.

Reply:
done